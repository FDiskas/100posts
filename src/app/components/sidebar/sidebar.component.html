<mat-sidenav-container class="sidenav-container">
  <mat-sidenav #drawer class="sidenav" fixedInViewport mode="over" [opened]="false">
    <mat-toolbar i18n>Navigation</mat-toolbar>
    <mat-nav-list *ngFor="let menuItem of menuItems">
      <a
        mat-list-item
        color="primary"
        [routerLink]="menuItem.path"
        routerLinkActive="active"
        (click)="drawer.toggle()"
        >{{ menuItem.name }}</a
      >
    </mat-nav-list>
  </mat-sidenav>
  <mat-sidenav-content fixedInViewport>
    <mat-toolbar color="primary">
      <button type="button" aria-label="Toggle sidenav" mat-icon-button (click)="drawer.toggle()">
        <mat-icon aria-label="Side nav toggle icon">menu</mat-icon>
      </button>
      <span>{{ title }}</span>
      <span class="spacer"></span>

      <button mat-button [matMenuTriggerFor]="menu"><mat-icon>language</mat-icon></button>
      <mat-menu #menu="matMenu">
        <a *ngFor="let locale of localesList" [href]="locale.code" mat-menu-item>{{ locale.label }}</a>
      </mat-menu>
    </mat-toolbar>
    <router-outlet></router-outlet>
  </mat-sidenav-content>
</mat-sidenav-container>
