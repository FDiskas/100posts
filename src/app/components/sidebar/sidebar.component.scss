:host {
  .sidenav-container {
    height: 100%;
  }

  .sidenav {
    width: 300px;
  }

  .sidenav .mat-toolbar {
    background: inherit;
  }

  .mat-toolbar.mat-primary {
    position: sticky;
    top: 0;
    z-index: 1;
  }
  .active {
    background-color: #3f51b5;
    color: #fff;
  }
  .spacer {
    flex: 1 1 auto;
  }
}
