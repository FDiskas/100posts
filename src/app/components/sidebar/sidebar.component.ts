import { Component, Input } from '@angular/core';
import { routes } from 'app/app-routing.module';

@Component({
  selector: 'app-side-bar',
  templateUrl: './sidebar.component.html',
  styleUrls: ['./sidebar.component.scss'],
})
export class SidebarComponent {
  @Input() title: string;

  @Input() localesList: { code: string; label: string }[] = [];

  menuItems = routes.filter((menu) => menu.data).map((item) => ({ path: item.path, name: item.data.name }));

  changeLanguage = (value: string) => {
    console.log(value);
    // window.location.href = value;
  };
}
