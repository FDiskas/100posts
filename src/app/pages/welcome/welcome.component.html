<div class="grid-container">
  <h1 class="mat-h1" i18n>How to use</h1>
  <p i18n>Welcome to the Posts Angular assignment. Please click Menu -> Posts to proceed</p>
</div>
