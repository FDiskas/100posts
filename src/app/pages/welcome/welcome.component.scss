:host {
  .grid-container {
    margin: 20px;
  }
}
