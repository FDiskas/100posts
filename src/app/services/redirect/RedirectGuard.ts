import { Injectable } from '@angular/core';

import { CanActivate, ActivatedRouteSnapshot } from '@angular/router';

@Injectable({
  providedIn: 'root',
})
export class RedirectGuard implements CanActivate {
  canActivate(route: ActivatedRouteSnapshot): boolean {
    Object.assign(document.createElement('a'), {
      target: '_blank',
      href: route.data.externalUrl,
    }).click();

    return true;
  }
}
