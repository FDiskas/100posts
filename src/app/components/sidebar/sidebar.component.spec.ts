import { LayoutModule } from '@angular/cdk/layout';
import { waitForAsync, ComponentFixture, TestBed } from '@angular/core/testing';
import { BrowserAnimationsModule, NoopAnimationsModule } from '@angular/platform-browser/animations';
import { MatButtonModule } from '@angular/material/button';
import { MatIconModule } from '@angular/material/icon';
import { MatListModule } from '@angular/material/list';
import { MatSidenavModule } from '@angular/material/sidenav';
import { MatToolbarModule } from '@angular/material/toolbar';
import { MatMenuModule } from '@angular/material/menu';
import '@angular/localize/init';
import { routes } from 'app/app-routing.module';

import { RouterModule } from '@angular/router';
import { WelcomeComponent } from 'app/pages/welcome/welcome.component';
import { APP_BASE_HREF } from '@angular/common';
import { RedirectGuardComponent } from 'app/services/redirect/RedirectGuard';
import { SidebarComponent } from './sidebar.component';

describe('SidebarComponent', () => {
  let component: SidebarComponent;
  let fixture: ComponentFixture<SidebarComponent>;

  beforeEach(
    waitForAsync(() => {
      TestBed.configureTestingModule({
        declarations: [SidebarComponent, WelcomeComponent, RedirectGuardComponent],
        imports: [
          BrowserAnimationsModule,
          NoopAnimationsModule,
          LayoutModule,
          MatButtonModule,
          MatIconModule,
          MatListModule,
          MatSidenavModule,
          MatToolbarModule,
          RouterModule,
          MatMenuModule,
          RouterModule.forRoot(routes),
        ],
        providers: [{ provide: APP_BASE_HREF, useValue: '/' }],
      }).compileComponents();
    }),
  );

  beforeEach(() => {
    // shallow = new Shallow(MyComponent, MyModule);
    fixture = TestBed.createComponent(SidebarComponent);

    component = fixture.componentInstance;
    component.localesList = [{ location: 'test', label: 'test' }];
    component.menuItems = [{ path: 'test', name: 'test' }];
    component.title = 'test';
    fixture.detectChanges();
  });

  it('should compile', () => {
    expect(component).toBeTruthy();
  });

  it('should contain menu items', () => {
    expect(component).toMatchSnapshot();
  });
});
