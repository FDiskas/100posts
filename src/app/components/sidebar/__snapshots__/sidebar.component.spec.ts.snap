// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SidebarComponent should contain menu items 1`] = `
SidebarComponent {
  "localesList": Array [
    Object {
      "label": "test",
      "location": "test",
    },
  ],
  "menuItems": Array [
    Object {
      "name": "test",
      "path": "test",
    },
  ],
  "title": "test",
}
`;
